packages/datadog-plugin-aws-sdk/src/services/sqs.js

```diff
@@ -11,6 +11,38 @@ class Sqs {
       'aws.sqs.queue_name': params.QueueName || params.QueueUrl
     })
   }
+
+  responseExtract (params, operation, response, tracer) {
+    if (operation === 'receiveMessage') {
+      if (
+        (!params.MaxNumberOfMessages || params.MaxNumberOfMessages === 1) &&
+        response &&
+        response.Messages &&
```

Comment:
The presence of a message should not change the behavior.

Reply:
What do you mean? `MessageAttributes` are store in `Message` so yeah we're going to need a `Message` to do anything here, no?

Reply:
What I mean is that the current code will fallback to the old behavior instead of `null` when there are no messages. Although I'm also not sure how there could be no messages.

Reply:
> What I mean is that the current code will fallback to the old behavior instead of null when there are no messages.

That's what we want, isn't it?

> Although I'm also not sure how there could be no messages.

When there are no messages in the queue, and you call `receiveMessage`, the `Messages` property will not be there at all, rather than being an empty array. At least, this is how localstack behaves.

Reply:
> That's what we want, isn't it?

It should explicitly set the parent to `null` to preserve the behavior for when propagation is supported.

> When there are no messages in the queue, and you call receiveMessage, the Messages property will not be there at all, rather than being an empty array. At least, this is how localstack behaves.

Should there even be a span at all in that case? I'm not sure how to deal with pulling since technically the "no message" case could not exist in a push model.

Reply:
> It should explicitly set the parent to null to preserve the behavior for when propagation is supported.

There's no new span to set the parent on. See below.

> Should there even be a span at all in that case? I'm not sure how to deal with pulling since technically the "no message" case could not exist in a push model.

Right, in that case there should not be a span at all, and instead we just propagate scope. See [this if-block](https://github.com/DataDog/dd-trace-js/pull/963/commits/e15b5371d0dd896306e8d73d074604b3ca041a30#diff-5477a55323f98152ec99122e4240cd4bR76-R84).